### PR TITLE
KAFKA-14691; [1/N] Add new fields to OffsetFetchRequest and OffsetFetchResponse - WIP

### PR DESCRIPTION
**Pull request by clolov**
This is part 1 of [KAFKA-14691](https://issues.apache.org/jira/browse/KAFKA-14691). I change the JSON files which are used to generate the POJOs used during requests and responses.

### CODE REVIEW

File: clients/src/main/resources/common/message/OffsetFetchRequest.json [hunk 1]
```diff
@@ -51,8 +57,9 @@
         "about": "The group ID."},
       { "name": "Topics", "type": "[]OffsetFetchRequestTopics", "versions": "8+", "nullableVersions": "8+",
         "about": "Each topic we would like to fetch offsets for, or null to fetch offsets for all topics.", "fields": [
-        { "name": "Name", "type": "string", "versions": "8+", "entityType": "topicName",
+        { "name": "Name", "type": "string", "versions": "0-8", "ignorable": true, "entityType": "topicName",
           "about": "The topic name."},
+        { "name": "TopicId", "type": "uuid", "versions": "9+", "ignorable": true, "about": "The unique topic ID" },
```
Comment (Hangleton) [inline clients/src/main/resources/common/message/OffsetFetchRequest.json]: nit: The `"about"` field could be on a new line.

File: clients/src/main/resources/common/message/OffsetFetchRequest.json [hunk 2]
```diff
@@ -33,11 +33,17 @@
...
   // Version 8 is adding support for fetching offsets for multiple groups at a time
-  "validVersions": "0-8",
+  //
+  // Version 9 adds GenerationIdOrMemberEpoch, MemberId and TopicId fields (KIP-848).
+  "validVersions": "0-9",
   "flexibleVersions": "6+",
   "fields": [
     { "name": "GroupId", "type": "string", "versions": "0-7", "entityType": "groupId",
       "about": "The group to fetch offsets for." },
+    { "name": "GenerationIdOrMemberEpoch", "type": "int32", "versions": "9+", "default": "-1", "ignorable": true,
```
Comment (Hangleton) [inline clients/src/main/resources/common/message/OffsetFetchRequest.json]: Do we want to add these fields in this PR?

**Review by Hangleton — COMMENTED** [review]
Thanks @clolov for the PR. Also tagging @jolshan who had started to look at the same a while ago.

**Comment by jolshan** [comment]
Since this is still under development, can we set the flag to indicate the latest version is unstable?

**Comment by dajac** [comment]
I have a few comments/questions:
* I am not really comfortable with merging this without the server side implementation. @clolov Is there a strong reason to not do them together?
* I agree with @Hangleton that it may be better to start with adding the TopicId only. This is complicated enough on its own. We can the other fields afterwards.
* I agree with @jolshan that we should set `"latestVersionUnstable": true` while in development.

**Comment by clolov** [comment]
> I have a few comments/questions:
> 
>     1. I am not really comfortable with merging this without the server side implementation. @clolov Is there a strong reason to not do them together?
> 
>     2. I agree with @Hangleton that it may be better to start with adding the TopicId only. This is complicated enough on its own. We can the other fields afterwards.
> 
>     3. I agree with @jolshan that we should set `"latestVersionUnstable": true` while in development.

I will accommodate 2 and 3 in subsequent commits. There was no reason for 1 other than splitting the pull request into nicer to review chunks. I am happy with trying to put Request/Respone changes + server side changes + tests in the same pull request similar to https://github.com/apache/kafka/pull/13240.

Thank you all for the reviews!

**Comment by dajac** [comment]
Closing this PR for now as the topic id work will be done later. We can re-open it when we resume the work.

**Pull request closed without merging**